File: src/main/java/social/alone/server/common/config/KakaoConfig.java
package social.alone.server.common.config;

import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import javax.validation.constraints.NotEmpty;

@Getter
@Configuration
@ConfigurationProperties(prefix = "app.kakao")
public class KakaoConfig {
    @NotEmpty
    private String apiKey;

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }
}

File: src/main/kotlin/social/alone/server/config/KakaoConfig.kt
package social.alone.server.config

import lombok.Getter
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

import javax.validation.constraints.NotEmpty

@Getter
@Configuration
@ConfigurationProperties(prefix = "app.kakao")
class KakaoConfig {
    @NotEmpty
    var apiKey: String? = null

    @NotEmpty
    var searchKeywordUrl: String? = null
}
